File: project/CCNet.VBDemos.Plugin/Tasks/CallHelloWorldTask.vb
Imports System.Diagnostics
Imports Exortech.NetReflector
Imports ThoughtWorks.CruiseControl.Core
Imports ThoughtWorks.CruiseControl.Core.Tasks

Public Class CallHelloWorldTask
    Inherits BaseExecutableTask

    Private myPersonsName As String
    <ReflectorProperty("name")> _
    Public Property PersonsName() As String
        Get
            Return myPersonsName
        End Get
        Set(ByVal value As String)
            myPersonsName = value
        End Set
    End Property

    Private myExecutable As String
    <ReflectorProperty("executable")> _
    Public Property Executable() As String
        Get
            Return myExecutable
        End Get
        Set(ByVal value As String)
            myExecutable = value
        End Set
    End Property

    Protected Overrides Function Execute(ByVal result As IIntegrationResult) As Boolean
        Dim processResult = TryToRun(CreateProcessInfo(result), result)
        result.AddTaskResult(New ProcessTaskResult(processResult))
        Return Not processResult.Failed
    End Function

    Protected Overrides Function GetProcessArguments(ByVal result As IIntegrationResult) As String
        Return """" & myPersonsName & """"
    End Function

    Protected Overrides Function GetProcessBaseDirectory(ByVal result As IIntegrationResult) As String
        Return result.WorkingDirectory
    End Function

    Protected Overrides Function GetProcessFilename() As String
        If String.IsNullOrEmpty(myExecutable) Then
            Return "HelloWorldApp"
        Else
            Return myExecutable
        End If
    End Function

    Protected Overrides Function GetProcessPriorityClass() As ProcessPriorityClass
        Return ProcessPriorityClass.Normal
    End Function

    Protected Overrides Function GetProcessTimeout() As Integer
        Return 300
    End Function

End Class

File: project/CCNet.CSharpDemos.Plugin/Tasks/CallHelloWorldTask.cs
using System.Diagnostics;
using Exortech.NetReflector;
using ThoughtWorks.CruiseControl.Core;
using ThoughtWorks.CruiseControl.Core.Tasks;

namespace CCNet.CSharpDemos.Plugin.Tasks
{
    [ReflectorType("callHelloWorld")]
    public class CallHelloWorldTask
        : BaseExecutableTask
    {
        [ReflectorProperty("name")]
        public string PersonsName { get; private set; }

        [ReflectorProperty("executable")]
        public string Executable { get; private set; }

        protected override string GetProcessFilename()
        {
            return string.IsNullOrEmpty(this.Executable) ?
                "HelloWorldApp" :
                this.Executable;
        }

        protected override string GetProcessArguments(IIntegrationResult result)
        {
            return "\"" + this.PersonsName + "\"";
        }

        protected override string GetProcessBaseDirectory(IIntegrationResult result)
        {
            return result.WorkingDirectory;
        }

        protected override ProcessPriorityClass GetProcessPriorityClass()
        {
            return ProcessPriorityClass.Normal;
        }

        protected override int GetProcessTimeout()
        {
            return 300;
        }

        protected override bool Execute(IIntegrationResult result)
        {
            var processResult = this.TryToRun(
                this.CreateProcessInfo(result),
                result);
            result.AddTaskResult(new ProcessTaskResult(processResult));
            return !processResult.Failed;
        }
    }
}
